docs/proposals/proposal-audit-log.md

```diff
@@ -0,0 +1,256 @@
+| TARGET_CREATED                 | User **john@acme.com** created a target named **Project Alpha**.                                          |
+| ROLE_CREATED                   | Admin **admin@acme.com** created a new role **SAMPLE_ROLE**.                                              |
+| ROLE_ASSIGNED                  | Admin **admin@acme.com** assigned a new role to user **john@acme.com**.                                   |
+| USER_REMOVED                   | Admin **admin@acme.com** removed user **john@acme.com**.                                                  |
+| ORG_TRANSFERRED                | Admin **admin@acme.com** transferred ownership.                                                           |
+| SCHEMA_CHECKED                 | CI made a schema check for **Project Alpha**.                                                             |
+| SCHEMA_PUBLISH                 | User **john@acme.com** published a new schema version for **Project Alpha**.                              |
+| SCHEMA_DELETED                 | Hive background job deleted old schema.                                                                   |
+| PROJECT_SETTINGS_UPDATED       | Changes made to project **Acme API** settings by **admin@acme.com**.                                      |
+| ORGANIZATION_SETTINGS_UPDATED  | Changes made to organization **Acme** settings by **admin@acme.com**.                                     |
+| TARGET_SETTINGS_UPDATED        | Changes made to target **Acme dev API** settings by **admin@acme.com**.                                   |
+| SCHEMA_POLICY_SETTINGS_UPDATED | Changes made to schema policy settings under an org or a project by **admin@acme.com**.                   |
+
+
+## Audit log UI
+
+Download would be triggered by a `Download CSV` button in Hive UI.
+
+We would provide these filters:
+- **date range** - to avoid pulling entire history of events
+- **by user** - in case somebody wants to see actions made by a user
+- **by project** - to scope down the logs to a specific project (by default audit log contains data from all projects of the org)
+- **by target** - to scope down it even more
+- **by action** - type or group of action types
+
+
+### Graphql API for UI
+
+Graphql Schema to support a UI with these filters looks like this:
+
+```graphql
+
+  scalar Date
+  scalar JSON
+
+  enum AuditLogEventAction {
+    USER_INVITED
+    USER_JOINED
+    EXPIRED_INVITE_HIT
+    PROJECT_CREATED
+    TARGET_CREATED
+    ROLE_CREATED
+    ROLE_ASSIGNED
+    USER_REMOVED
+    ORG_TRANSFERRED
+    SCHEMA_CHECKED
+    SCHEMA_PUBLISH
+    SCHEMA_DELETED
+    PROJECT_SETTINGS_UPDATED
+    ORGANIZATION_SETTINGS_UPDATED
+    TARGET_SETTINGS_UPDATED
+    SCHEMA_POLICY_SETTINGS_UPDATED
+  }
+
+  type AuditLog {
+    event_time: Date!
+    user_id: String
+    user_email: String
+    organization_id: String
```

Comment:
```suggestion
    organization_id: String!
```

I think we will always have organization_id

Reply:
correct. We can always make it non nullable if we introduce event for something like user signup